oxd-server/src/main/java/org/gluu/oxd/server/mapper/RegisterRequestMapper.java

```diff
@@ -0,0 +1,410 @@
+package org.gluu.oxd.server.mapper;
+
+import com.google.common.base.Strings;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang.math.NumberUtils;
+import org.gluu.oxauth.client.RegisterRequest;
+import org.gluu.oxauth.model.common.AuthenticationMethod;
+import org.gluu.oxauth.model.common.GrantType;
+import org.gluu.oxauth.model.common.ResponseType;
+import org.gluu.oxauth.model.common.SubjectType;
+import org.gluu.oxauth.model.crypto.encryption.BlockEncryptionAlgorithm;
+import org.gluu.oxauth.model.crypto.encryption.KeyEncryptionAlgorithm;
+import org.gluu.oxauth.model.crypto.signature.SignatureAlgorithm;
+import org.gluu.oxauth.model.register.ApplicationType;
+import org.gluu.oxd.server.service.Rp;
+
+import java.util.stream.Collectors;
+
+public class RegisterRequestMapper {
+
+    public void fillRp(Rp rp, RegisterRequest request) {
+
+        if (!Strings.isNullOrEmpty(request.getClientName())) {
+            rp.setClientName(request.getClientName());
+        }
+
+        if (request.getApplicationType() != null) {
+            rp.setApplicationType(request.getApplicationType().toString());
+        }
+
+        if (request.getTokenEndpointAuthSigningAlg() != null && StringUtils.isNotBlank(request.getTokenEndpointAuthSigningAlg().toString())) {
+            rp.setTokenEndpointAuthSigningAlg(request.getTokenEndpointAuthSigningAlg().toString());
+        }
+
+        if (request.getGrantTypes() != null) {
+            rp.setGrantType(request.getGrantTypes().stream().map(item -> item.getValue()).collect(Collectors.toList()));
+        }
+
+        if (request.getFrontChannelLogoutUris() != null) {
+            rp.setFrontChannelLogoutUris(request.getFrontChannelLogoutUris());
+        }
+
+        if (request.getTokenEndpointAuthMethod() != null && StringUtils.isNotBlank(request.getTokenEndpointAuthMethod().toString())) {
+            rp.setTokenEndpointAuthMethod(request.getTokenEndpointAuthMethod().toString());
+        }
+
+        if (request.getRequestUris() != null && !request.getRequestUris().isEmpty()) {
+            rp.setClientRequestUris(request.getRequestUris());
+        }
+
+        if (!Strings.isNullOrEmpty(request.getSectorIdentifierUri())) {
+            rp.setClientSectorIdentifierUri(request.getSectorIdentifierUri());
+        }
+
+        if (request.getRedirectUris() != null && !request.getRedirectUris().isEmpty()) {
+            rp.setRedirectUris(request.getRedirectUris());
+            rp.setRedirectUri(request.getRedirectUris().get(0));
+        }
+
+        if (request.getAccessTokenAsJwt() != null) {
+            rp.setAccessTokenAsJwt(request.getAccessTokenAsJwt());
+        }
+
+        if (request.getAccessTokenSigningAlg() != null && StringUtils.isNotBlank(request.getAccessTokenSigningAlg().toString())) {
+            rp.setAccessTokenSigningAlg(request.getAccessTokenSigningAlg().toString());
+        }
+
+        if (request.getRptAsJwt() != null) {
```

Comment:
This if seems to be redundant, if its null then we set null which is ok.

Reply:
Changes done